### PR TITLE
import setting form fixes

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

**Comment by DhaaraniCIT** [comment]
full configuration import field
Screenshot 2023-11-06 at 3 48 56 PM

File: src/app/core/models/common/import-settings.model.ts [hunk 1]
```diff
@@ -19,4 +19,11 @@ export type ImportSettingMappingRow = {
   is_custom: boolean,
   source_field: string,
   source_placeholder: string | null
+}
+
+export type customField = {
```
Comment (ashwin1111) [inline src/app/core/models/common/import-settings.model.ts:24]: start with caps, ImportSettingsCustomFieldRow

File: src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts [hunk 2]
```diff
@@ -102,13 +109,17 @@ export class Sage300ImportSettingModel {
       }
 
     static mapAPIResponseToFormGroup(importSettings: Sage300ImportSettingGet | null, sage300Fields: IntegrationField[]): FormGroup {
+        const importsettings = importSettings ? importSettings : undefined;
```
Comment (ashwin1111) [inline src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:112]: not needed, send null or importSettings from component

File: src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts [hunk 3]
```diff
@@ -102,13 +109,17 @@ export class Sage300ImportSettingModel {
...
+            expenseFields: new FormArray(expenseFieldsArray),
             isDependentImportEnabled: new FormControl(importSettings?.dependent_field_settings?.is_import_enabled ? importSettings.dependent_field_settings.is_import_enabled : false),
             costCodes: new FormControl(importSettings?.dependent_field_settings?.cost_code_field_name ? this.generateDependentFieldValue(importSettings.dependent_field_settings.cost_code_field_name, importSettings.dependent_field_settings.cost_code_placeholder) : null),
-            costCategory: new FormControl(importSettings?.dependent_field_settings?.cost_category_field_name ? this.generateDependentFieldValue(importSettings.dependent_field_settings.cost_category_field_name, importSettings.dependent_field_settings.cost_category_placeholder) : null)
+            costCategory: new FormControl(importSettings?.dependent_field_settings?.cost_category_field_name ? this.generateDependentFieldValue(importSettings.dependent_field_settings.cost_category_field_name, importSettings.dependent_field_settings.cost_category_placeholder) : null),
+            dependentFieldImportToggle: new FormControl(true)
```
Comment (ashwin1111) [inline src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:122]: this should be read from importSettings.dependent_field_settings.is_enabled

File: src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.html [hunk 4]
```diff
@@ -1,2 +1,2 @@
 <p>sage300-import-settings works!</p>
-<app-configuration-import-field [form]="importSettingForm" [appName]="appName" [defaultImportFields]="defaultImportFields" [costCategoryOption]="costCategoryOption" [costCodeFieldOption]="costCodeFieldOption" [accountingFieldOptions]="sage300Fields" [fyleFieldOptions]="fyleFields" [form]="importSettingForm"></app-configuration-import-field>
+<app-configuration-import-field [form]="importSettingForm" [appName]="appName" [defaultImportFields]="defaultImportFields" [costCategoryOption]="costCategoryOption" [costCodeFieldOption]="costCodeFieldOption" [accountingFieldOptions]="sage300Fields" [fyleFieldOptions]="fyleFields" [form]="importSettingForm" [dependentImportFields]="dependentImportFields"></app-configuration-import-field>
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.html:2]: add all input and output as separate lines, for better readability

File: src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts [hunk 5]
```diff
@@ -70,30 +89,83 @@ export class Sage300ImportSettingsComponent implements OnInit {
     private helper: HelperService
   ) { }
 
+  saveCustomField(customFieldForms: FormGroup, type: string) {
```
Comment (ashwin1111) [inline src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts:92]: split if else to separate functions, looks too long

File: src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts (hunk 3 above)
Comment (DhaaraniCIT) [reply src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:122]: this is for the form field, it is like a dummy one
`<p-inputSwitch class="tw-pl-32-px input-toggle-section tw-pt-10-px" formControlName="dependentFieldImportToggle" [disabled]="true"></p-inputSwitch`

Comment (DhaaraniCIT) [reply src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:122]: it is disabled in the form in component

File: src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts (hunk 5 above)
Comment (DhaaraniCIT) [reply src/app/integrations/sage300/sage300-shared/sage300-import-settings/sage300-import-settings.component.ts:92]: this is fixed in next PR (https://github.com/fylein/fyle-integrations-settings-app/pull/246)

File: src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts (hunk 2 above)
Comment (DhaaraniCIT) [reply src/app/core/models/sage300/sage300-configuration/sage300-import-settings.model.ts:112]: this is fixed in this PR (https://github.com/fylein/fyle-integrations-settings-app/pull/246)

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**